### DISCUSSION TITLE
async execute select with a where case just stalling am i doing something incorrectly?

### DISCUSSION THREAD

**Discussion by karlohrn** [psycopg/psycopg#354] [Q&A]
Using psycopg_pool asyncConnectionPool and trying to execute a "select * from table where name='test'" 
psycopg-pool==3.1.1
Python 3.9.10
mac m1

code:
```
# self.__pool = psycopg_pool.AsyncConnectionPool(connection_info)
# and I have used it for other operations like inserts with success.  
async with self.__pool.connection() as connection: 
     async with connection.cursor() as cursor:
         query = sql.SQL("SELECT company FROM {} WHERE companyid={};").format(
             sql.Identifier("companies"), sql.Identifier(companyid))
         await cursor.execute(query)
         # BUT NEVER REACH THIS POINT IT JUST HANGS
         return await cursor.fetchone()
```
The problem I have is that the await cursor.execute(query) just hangs and I never finishes. 

The table I am going to selecting from is empty so i where expecting a None. 

And if i remove the "where" statment from the selection the curor.execute coroutine resolves fast.  

Am I missing something?

**Comment by dvarrazzo** [comment]
Maybe there is some row locked in the database?

**Reply by karlohrn** [reply]
I found the problem. The fault was my worker and query did not handle psycopg.errors.UndefinedColumn so once i expected the Exceptions it resolved the query. But i was a but surprised that the exception did not crash the run just stalled until i put it in a try except block.

**Reply by dvarrazzo** [reply]
That doesn't sound correct. Can you provide a case to reproduce it?
